jikanpy/abstractjikan.py

```diff
@@ -51,25 +50,47 @@ def __init__(
         self.season_later_url: str = selected_base + "season/later"
 
     def _check_response(
-        self, response: Any, **kwargs: Union[int, Optional[str]]
+        self,
+        response_dict: Dict,
+        response_status_code: int,
+        **kwargs: Union[int, Optional[str]]
     ) -> None:
         """
         Check if the response is an error
 
         Keyword Arguments:
-        response -- response from the API call
+        response_dict -- parsed response from the API call
+                         is empty ({}) when there was a json.decoder.JSONDecodeError
+        response_status -- the corresponding http code for the response
         kwargs -- keyword arguments
         """
-        if response.status_code >= 400:
-            try:
-                json_resp = response.json()
-                error_msg = json_resp.get("error")
-            except json.decoder.JSONDecodeError:
-                error_msg = ""
-            err_str: str = "{} {}: error for ".format(response.status_code, error_msg)
+        if response_status_code >= 400:
+            error_msg = response_dict.get("error", "")
+            err_str: str = "{} {}: error for ".format(response_status_code, error_msg)
             err_str += ", ".join("=".join((str(k), str(v))) for k, v in kwargs.items())
             raise APIException(err_str)
 
+    def _add_jikan_metadata(
+        self,
+        response: Any,  # Union[requests.Response, aiohttp.ClientResponse]
```

Comment:
Is there a reason that we put the type annotation in a comment instead of the below?
```python
response: Union[requests.Response, aiohttp.ClientResponse]
```

Reply:
This was an artifact of when I originally was overriding _add_jikan_metadata in the subclasses, this can be changed.